Adds two new traumas, Mind Echo and Existential Crisis

XDTM opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds two new traumas in the trauma pool.
Mind Echo: a mild trauma that records what the patient hears and says, and occasionally replaces heard or spoken messages with one of the recorded ones.
Existential Crisis: a special trauma that causes spontaneous temporary non-existence, which works like a desynchronizer does.

## Why It's Good For The Game

More variety in traumas!

## Changelog
:cl:
add: Added two new traumas in the trauma pool: Mind Echo (mild) and Existential Crisis (special).
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

XDTM committed:
Adds two new traumas

tralezab commented:
Existential Crisis is an amazing name for it though, that's sick.

ATH1909 commented:
anconfusedrock kind of has a point, tbh. While I think that mind echo is unique enough, is there anything that we can change about existential crisis to make it more unique?

OverlyGeneric commented:
> anconfusedrock kind of has a point, tbh. While I think that mind echo is unique enough, is there anything that we can change about existential crisis to make it more unique?

Be severely depressed and slowly fade away the sadder you are.

tralezab commented:
Having your mood change how "real" you are and a low mood making you not exist anymore would be very cool

Grazyn commented:
There is an actual mental illness in which you believe you don't exist, it's called Cotard's delusion.

XDTM commented:
Now i'm torn between the wordplay name or the obscure based-in-reality one...
Although reading the wiki page it can lead to delusions of immortality, which can be a nice idea for a separate special trauma in some form

ATH1909 commented:
> Although reading the wiki page it can lead to delusions of immortality, which can be a nice idea for a separate special trauma in some form

We already have anosognosia, just so you know.

Grazyn commented:
I admit, existential crisis is way funnier, also I advise against using the name cotard for a "immortality delusion" trauma because Cotard's is mostly known as the "delusion of death" thing in popular culture, despite the actual article saying that immortality is the majority

ExcessiveUseOfCobblestone commented:
is existential crisis just the immortality talisman

davethwave commented:
Thought of a fun feature, how about if you use the desynchronizer too long for perhaps 7 minutes total or something you gain the existential crisis trauma?

XDTM commented:
If we do that it might be worth doing similar stuff for other traumas, like gaining phobias from certain events, as long as it's a rare event.
@ExcessiveUseOfCobblestone i guess, but it's completely incontrollable as well as invisible, and the patient can do stuff while not-existing, like when using a desynchronizer

ExcessiveUseOfCobblestone commented:
I see...

ExcessiveUseOfCobblestone commented on code/datums/brain_damage/special.dm:
```diff
@@ -182,8 +182,56 @@
+/datum/brain_trauma/special/existential_crisis/on_life()
+	..()
+	if(!veil && world.time > next_crisis && prob(3))
+		fade_out()
```
This issue will need to be ratified

XDTM replied:
i'll have some more time to code this weekend, i'll make sure to prevent those cases

ExcessiveUseOfCobblestone commented:
Update?

XDTM committed:
only public nonexistence

XDTM commented:
Made it so it only triggers when loc is a turf, should get rid of buggy interactions with containers altogether.

AnturK commented on code/datums/brain_damage/mild.dm:
```diff
@@ -226,3 +226,39 @@
+	if(hear_dejavu.len >= 5)
+		if(prob(15))
+			var/deja_vu = pick_n_take(hear_dejavu)
+			var/new_message = replacetext(message, regex("\".+\"", "gi"),"\"[deja_vu]\"") //Quotes included to avoid cases where someone says part of their name
```
Should store this regex since it's static.

AnturK commented on code/datums/brain_damage/special.dm:
```diff
@@ -182,8 +182,57 @@
+/datum/brain_trauma/special/existential_crisis/proc/fade_out()
+	if(veil)
+		return
+	var/duration = pick(50, 450)
```
Was this supposed to be rand or it's correct ?

XDTM replied:
oops, yeah it's rand

AnturK reviewed (changes requested):
Just a nitpick

XDTM commented:
@optimumtact i found this old PR in my list; i fixed the conflicts and addressed the AnturK review, and it's functional. 

This PR can be reopened, or i could open another one from the branch if it's an issue.

ExcessiveUseOfCobblestone commented:
Old PRs regarding balance/features should be re-PR'd (preferably with link to old) so it shows on the servers.

XDTM commented:
Alright, will do
